### PR TITLE
OTLP+Arrow Receiver Test

### PR DESCRIPTION
**Pull request by jmacd**
**Description:** Add a basic test for receiving OTLP+Arrow traces and logs.

**Testing:** See also https://github.com/f5/otel-arrow-adapter/pull/44

This gets basic coverage of the `internal/arrow` package. It does not test error responses at this time, just the success path.

### CODE REVIEW

**Review by lquerel — APPROVED** [review]
No comment. LGTM

**Review by paivagustavo — APPROVED** [review]
LGTM, one small remark about some copied protos not being used.

File: receiver/otlpreceiver/internal/arrow/arrow_test.go [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+	}
+}
+
+func (tc *testChannel) doAndReturnConsumeTraces(normalReturn error) func(ctx context.Context, traces ptrace.Traces) error {
+	return func(ctx context.Context, traces ptrace.Traces) error {
+		tc.rlock.Lock()
+		defer tc.rlock.Unlock()
+		copy := ptrace.NewTraces()
+		traces.CopyTo(copy)
+		tc.recvTraces = append(tc.recvTraces, traces)
```
[the remark this answers is not in the document]
Comment (jmacd) [reply receiver/otlpreceiver/internal/arrow/arrow_test.go]: Thanks!

File: receiver/otlpreceiver/internal/arrow/arrow_test.go [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+	err = stop()
+	require.Error(t, err)
+	require.True(t, errors.Is(err, context.Canceled))
+
+	// EqualValues works for the underlying gogo protos.
+	assert.EqualValues(t, tc.recvTraces, []ptrace.Traces{td})
+
+	// cmp.Diff works for new-style google protobuf protos.
+	require.Equal(t, "", cmp.Diff(tc.sentStatus, []*arrowpb.BatchStatus{
```
[the remark this answers is not in the document]
Comment (jmacd) [reply receiver/otlpreceiver/internal/arrow/arrow_test.go:272]: It's not well documented!

**Pull request merged**